noark-extraction-validator/src/main/java/com/documaster/validator/validation/noark53/validators/NoarkXMLHandler.java

```diff
@@ -0,0 +1,225 @@
+import com.documaster.validator.validation.utils.AbstractReusableXMLHandler;
+import org.apache.commons.lang.StringUtils;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXParseException;
+
+import static java.util.stream.Collectors.collectingAndThen;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * An {@link AbstractReusableXMLHandler} implementation that transforms the captures the raised {@link
+ * SAXParseException} during parsing, enriches them with element metadata information, and transforms them to {@link
+ * BaseItem}s.
+ */
+class NoarkXMLHandler extends AbstractReusableXMLHandler {
+
+	private Schema schema;
+	private String elementType;
+	private final Map<NoarkReferenceField, String> idFields = new LinkedHashMap<>();
+	private List<NoarkSAXParseException> exceptions = new ArrayList<>();
+	private StringBuilder characters = new StringBuilder();
+
+	NoarkXMLHandler(Schema schema) {
+
+		this.schema = schema;
+	}
+
+	private void setElementType(String qName, Attributes attributes) {
+
+		if (!NoarkElementType.contains(qName)) {
+			return;
+		}
+
+		elementType = null;
+		for (int i = 0; i < attributes.getLength(); i++) {
+			if (attributes.getQName(i).endsWith("type")) {
+				elementType = attributes.getValue(i);
+				break;
+			}
+		}
+
+		elementType = !StringUtils.isBlank(elementType) ? elementType : qName;
+	}
+
+	private void setFieldValue(NoarkReferenceField field, String value) {
+
+		idFields.put(field, StringUtils.trimToNull(value));
+	}
+
+	@Override
+	public boolean hasExceptions() {
+
+		return !exceptions.isEmpty();
+	}
+
+	@Override
+	public List<BaseItem> getExceptionsAsItems() {
+
+		return exceptions.stream()
+				.sorted(new SAXParseExceptionComparator())
+				.map(ex -> {
+
+					BaseItem item = new BaseItem();
+
+					item.add("Schema", schema.toString());
+					item.add("Type", ex.getElementType());
+					ex.getIdFields().forEach((k, v) -> item.add(k.toString(), v));
+					item.add("Line", ex.getLineNumber() != -1 ? ex.getLineNumber() : null);
+					item.add("Column", ex.getColumnNumber() != -1 ? ex.getColumnNumber() : null);
+					item.add("Message", ex.getMessage());
+
+					return item;
+				}).collect(collectingAndThen(toList(), Collections::unmodifiableList));
+	}
+
+	@Override
+	public void reset() {
+
+		exceptions.clear();
+		characters.setLength(0);
+		elementType = null;
+		Stream.of(NoarkReferenceField.values()).forEach(v -> idFields.put(v, null));
+	}
+
+	@Override
+	public void startElement(String uri, String localName, String qName, Attributes attributes) {
+
+		characters.setLength(0);
+		setElementType(qName, attributes);
+	}
+
+	@Override
+	public void characters(char buf[], int offset, int length) {
+
+		characters.append(new String(buf, offset, length));
+	}
+
+	@Override
+	public void endElement(String uri, String localName, String qName) {
+
+		for (NoarkReferenceField field : NoarkReferenceField.values()) {
+			if (qName.equalsIgnoreCase(field.getName())) {
```

Comment:
The _mappeID_, _registreringsID_, and _systemID_ fields, for example, are not defined as all-lowercase in _arkivstruktur.xsd_. By introducing lowercase-only field names in the _NoarkReferenceField_ enum and then performing _equalsIgnoreCase_ in the code leads to a smaller change of a human error when defining the Noark structure.